lib/zoneinfo_module.c

```diff
@@ -1204,7 +1207,8 @@ calendarrule_new(uint8_t month, uint8_t week, uint8_t day, int8_t hour,
         return -1;
     }
 
-    if (day < 0 || day > 6) {
+    // Unsigned data type means that `day` will never be less than 0
+    if (day > 6) {
```

Comment:
It would be great if we could have a compile-time assertion that this is an unsigned data type, since if this changes we won't be properly validating the inputs. That seems more dangerous in some ways than having `day < 0`, which will get optimized out in the right situation anyway.